streams/runners/storm/runtime/src/main/java/com/hortonworks/streamline/streams/runtime/storm/spout/AvroStreamsSnapshotDeserializer.java

```diff
@@ -86,15 +89,21 @@ private Object convertValue(Object deserializedObj) {
             Map<Object, Object> map = (Map<Object, Object>) deserializedObj;
             ImmutableMap.Builder<String, Object> keyValues = ImmutableMap.builder();
             for (Map.Entry entry : map.entrySet()) {
-                keyValues.put(entry.getKey().toString(), convertValue(entry.getValue()));
+                Object currentValue = convertValue(entry.getValue());
+                if (currentValue != null) {
+                    keyValues.put(entry.getKey().toString(), currentValue);
+                }
             }
             value = keyValues.build();
 
         } else if (deserializedObj instanceof Collection) { // type of array
             Collection<Object> collection = (Collection<Object>) deserializedObj;
             ImmutableList.Builder<Object> values = ImmutableList.builder();
             for (Object obj : collection) {
-                values.add(convertValue(obj));
+                Object currentValue = convertValue(obj);
+                if (currentValue != null) {
```

Comment:
We should not ignore null values for a field but the `values` can be changed to `ArrayList` and set the records value as unmodifiable list of the earlier ArrayList.